lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,451 @@
+        try:
+            response = self.traffic_manager_management_client.profiles.get(self.resource_group, self.name)
+            self.log("Response : {0}".format(response))
+            self.log("Traffic Manager profile : {0} found".format(response.name))
+            self.endpoints_copy = response.endpoints if response and response.endpoints else None
+            return traffic_manager_profile_to_dict(response)
+        except CloudError:
+            self.log('Did not find the Traffic Manager profile.')
+            return False
+
+    def delete_traffic_manager_profile(self):
+        '''
+        Deletes the specified Traffic Manager profile in the specified subscription and resource group.
+        :return: True
+        '''
+
+        self.log("Deleting the Traffic Manager profile {0}".format(self.name))
+        try:
+            operation_result = self.traffic_manager_management_client.profiles.delete(self.resource_group, self.name)
+            return True
+        except CloudError as e:
+            self.log('Error attempting to delete the Traffic Manager profile.')
+            self.fail("Error deleting the Traffic Manager profile: {0}".format(e.message))
+            return False
+
+    def ceate_update_traffic_manager_profile(self):
+        '''
+        Creates or updates a Traffic Manager profile.
+
+        :return: deserialized Traffic Manager profile state dictionary
+        '''
+        self.log("Creating / Updating the Traffic Manager profile {0}".format(self.name))
+
+        parameters = Profile(
+            tags=self.tags,
+            location=self.location,
+            profile_status=self.profile_status,
+            traffic_routing_method=self.routing_method,
+            dns_config=create_dns_config_instance(self.dns_config) if self.dns_config else None,
+            monitor_config=create_monitor_config_instance(self.monitor_config) if self.monitor_config else None,
+            endpoints=self.endpoints_copy
+        )
+        try:
+            response = self.traffic_manager_management_client.profiles.create_or_update(self.resource_group, self.name, parameters)
+            return traffic_manager_profile_to_dict(response)
+        except CloudError as exc:
+            self.log('Error attempting to create the Traffic Manager.')
+            self.fail("Error creating the Traffic Manager: {0}".format(exc.message))
+
+    def check_update(self, response):
+        if response['location'] != self.location:
+            self.log("Location Diff - Origin {0} / Update {1}".format(response['location'], self.location))
+            return True
+
+        if response['profile_status'].lower() != self.profile_status.lower():
+            self.log("Profile Status Diff - Origin {0} / Update {1}".format(response['profile_status'], self.profile_status))
+            return True
+
+        if response['routing_method'].lower() != self.routing_method.lower():
```

Comment:
You shouldn't need the `.lower()` on the `self.routing_method`, the module arg spec will ensure it is lower case based on the choices you set.

Reply:
updated